frontends/verilog/verilog_parser.y

```diff
@@ -610,26 +610,26 @@ non_opt_delay:
 delay:
 	non_opt_delay | %empty;
 
+io_wire_type:
+	wire_type_token_io wire_type_const_rand opt_wire_type_token wire_type_signed;
+
+not_io_wire_type:
+	wire_type_const_rand wire_type_token wire_type_signed;
+
+no_implicite_wire_type:
+	wire_type_const_rand wire_type_token wire_type_signed;
+
+wire_type_without_implicit:
+	{ astbuf3 = new AstNode(AST_WIRE); current_wire_rand = false; current_wire_const = false; }
+	no_implicite_wire_type { $$ = astbuf3; SET_RULE_LOC(@$, @2, @$); };
```

Comment:
I think I understand how these grammar changes resolve the `TOK_USERTYPE` conflict in the `typedef` rule, but I'm wondering if we can simplify it a little. Could we have both `io_wire_type` and `non_io_wire_type`, both of which set and use `astbuf3`, have `wire_type` be either of those, and then have the `typedef` rule use `non_io_wire_type` directly? This would eliminate `wire_type_without_implicit`, `no_implicite_wire_type`, and `wire_type_wrapper`.

Reply:
Fixed.